### DISCUSSION TITLE
Welcome to immersive-translate Discussions!

### DISCUSSION THREAD

**Discussion by theowenyoung** [immersive-translate/immersive-translate#27] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by vincehi** [comment]
Hello everyone,I just discovered this project and I love it already. I'd like to be able to participate but first I'll tell my story with the translation add-ons. I have always used `TWP` on Firefox and I find it incredible compared to the competition. What I love about it is that it translates what is visible in the display window, which allows it to be very fast. Moreover, it avoids the `<pre>` tags. And last but not least it follows my navigation and doesn't force me to press the button every time. But here I was, needing to switch back to safari and I was helpless without this add-ons. So I thought I'd port TWP to safari. Which I partially succeeded in doing and with the V3 manifest. I still have some bugs but most things work and without using userscripts, and without paying for an apple developper account (I could make a small implementation guide).
I find this project promising but some things still prevent me from doing without TWP for immersive-translate for the moment. Indeed, I prefer text replacement as TWP does. I also prefer the more elegant "native" safari extension solution rather than userscripts.
Do you share this view? And if yes I would be happy to contribute and bring the fruit of my experiments.
Thanks again for the work.

**Reply by theowenyoung** [reply]
Hi, @bazalp , Thanks for your message, I would like to explain the status of this extension.

Originally, this extension was not created for general translation, but to allow users to learn another language while reading bilingual information, so it is not so easy to hide the original text at present.

But I do plan to support the feature of only displaying translations, but I have not started to do it yet. We have a small development group, but the members are all Chinese-speaking (may cause inconvenience to you), and the source code is in a private project. Let me invite you first, you can look around first~

> most things work and without using userscripts, and without paying for an apple developer account (I could make a small implementation guide

I'm interested in how you did it, did you implement it in a way just for yourself, or is it convenient for everyone to do this? On the other hand, for this extended greasemonkey script, it has most of the functions, so even if there is no native extension, I don't think it will matter.

Thanks agian for your message.

**Reply by vincehi** [reply]
Thank you for the invitation, I am very grateful. 

I will write a guide to show you how I did it, but it is not easy for most people. Also I need to create a repository for the changes I made on TWP so I can use them as inspiration.

See you soon

**Reply by Spikee07** [reply]
Ok

**Comment by H-cnm** [comment]
手机可以翻译吗？

**Reply by aojunhao123** [reply]
您好，是可以的。
https://immersivetranslate.com/zh-Hans/
